### PR TITLE
Clarify mutually exclusive options in span creation.

### PR DESCRIPTION
**Pull request by songy23**
Fixed the comment in https://github.com/open-telemetry/opentelemetry-specification/pull/124#discussion_r294392248.

### CODE REVIEW

File: specification/tracing-api.md [hunk 1]
```diff
@@ -213,6 +213,15 @@ Optional parameters (or corresponding setters on `Builder` if using a `Builder`
 - The option to become a root `Span` for a new trace.
   If not set, the value of [Tracer.getCurrentSpan](#getcurrentspan) at `StartSpan`
   time will be used as parent.
+
+Note: The three parameters above (parent `Span`, parent `SpanContext` and `root`) are
+mutually exclusive. If multiple parameters are specified or corresponding `Setter`s are
+called multiple times, only the last specified value will be used. For example:
+1. `builder.setParent(parentSpan).setNoParent().startSpan()` will generate a new root span and
+`parentSpan` will be ignored;
```
Comment (SergeyKanzhelev) [inline specification/tracing-api.md]:
```suggestion
  `parentSpan` will be ignored;
```

File: specification/tracing-api.md [hunk 2]
```diff
@@ -213,6 +213,15 @@ Optional parameters (or corresponding setters on `Builder` if using a `Builder`
...
+2. `tracer.StartSpan(options.WithNoParent(), options.WithParentContext(parentCtx))`
+will generate a new child span with remote parent `parentCtx`, and `WithNoParent` will be ignored.
```
Comment (SergeyKanzhelev) [inline specification/tracing-api.md]:
```suggestion
  will generate a new child span with remote parent `parentCtx`, and `WithNoParent` will be ignored.
```

File: specification/tracing-api.md [hunk 3]
```diff
@@ -213,6 +213,15 @@ Optional parameters (or corresponding setters on `Builder` if using a `Builder`
 - The option to become a root `Span` for a new trace.
   If not set, the value of [Tracer.getCurrentSpan](#getcurrentspan) at `StartSpan`
   time will be used as parent.
+
+Note: The three parameters above (parent `Span`, parent `SpanContext` and `root`) are
+mutually exclusive. If multiple parameters are specified or corresponding `Setter`s are
```
Comment (reyang) [inline specification/tracing-api.md]: It feels like we assume there will be multiple parameters. Probably we can leave it as a language specific decision?

File: specification/tracing-api.md (hunk 1 above)
Comment (songy23) [reply specification/tracing-api.md]: Both fixed in https://github.com/open-telemetry/opentelemetry-specification/pull/131/commits/2d0eef2383d94b25b966349d679b6117b5141992.

File: specification/tracing-api.md (hunk 3 above)
Comment (songy23) [reply specification/tracing-api.md]: SG, added a note here.

**Pull request merged**